From a80a29e1cba6cdb37e62422b75caa2e7b6dc8c3e Mon Sep 17 00:00:00 2001
From: Sandipan Patra <spatra@nvidia.com>
Date: Tue, 23 May 2023 14:29:27 +0530
Subject: [PATCH 364/364] WAR: printk: remove syslog_print_all from do_syslog
X-NVConfidentiality: public

Removing call to syslog_print_all() in do_syslog() to avoid
the lockup.

 BUG: spinlock already unlocked on CPU#0, kerneloops/5158
  lock: logbuf_lock+0x0/0x18, .magic: dead4ead, .owner: <none>/-1,
.owner_cpu: -1
 CPU: 0 PID: 5158 Comm: kerneloops Not tainted 4.9.299-rt191-tegra #6
 Hardware name: quill (DT)
 Call trace:
 [<ffffff800808b350>] dump_backtrace+0x0/0x198
 [<ffffff800808b90c>] show_stack+0x24/0x30
 [<ffffff8008f72f84>] dump_stack+0xa0/0xc4
 [<ffffff800810e7c4>] spin_dump+0x84/0xb0
 [<ffffff800810ea1c>] do_raw_spin_unlock+0xc4/0x100
 [<ffffff8008f7e080>] _raw_spin_unlock_irq+0x20/0x68
 [<ffffff8008115ae0>] syslog_print_all+0x1a0/0x3d8
 [<ffffff8008116c08>] do_syslog+0xf0/0x550
 [<ffffff80081170a0>] SyS_syslog+0x38/0x50
 [<ffffff8008083100>] el0_svc_naked+0x34/0x38

Bug 3991287

Change-Id: If88547cb5e60d8271de821369903f6df5a41ad55
Signed-off-by: Sandipan Patra <spatra@nvidia.com>
---
 kernel/printk/printk.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index 3befe7f56193..1dcbee5a8ad4 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -1575,7 +1575,9 @@ int do_syslog(int type, char __user *buf, int len, int source)
 			error = -EFAULT;
 			goto out;
 		}
-		error = syslog_print_all(buf, len, clear);
+
+	/* WAR: Bug 3991287 */
+	/*	error = syslog_print_all(buf, len, clear); */
 		break;
 	/* Clear ring buffer */
 	case SYSLOG_ACTION_CLEAR:
-- 
2.17.1

